File: testdata/bench/kotlin/BenchmarkTest00010.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import java.sql.DriverManager

// Vulnerable: SQL injection via header value
suspend fun handler00010(call: ApplicationCall) {
    val tenant = call.request.headers["X-Tenant-ID"]
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    conn.createStatement().executeQuery("SELECT * FROM data WHERE tenant_id = '$tenant'")
    call.respondText("OK")
}

File: testdata/bench/go/BenchmarkTest00010.go
package bench

import (
	"database/sql"
	"net/http"
)

// SQL injection via Cookie
func Handler00010(w http.ResponseWriter, r *http.Request) {
	cookie, _ := r.Cookie("session_data")
	db, _ := sql.Open("sqlite3", ":memory:")
	db.Query("SELECT * FROM sessions WHERE data = '" + cookie.Value + "'")
}
